python/foglamp/plugins/south/modbustcp/modbustcp.py

```diff
@@ -26,49 +32,84 @@
     As an example of how to use this library:
 
         from pymodbus3.client.sync import ModbusTcpClient
+        
         client = ModbusTcpClient('127.0.0.1')
         value = (client.read_holding_registers(0, 1, unit=1)).registers[0]
         print(value)
         client.close()
 
     """
 
+import copy
 from datetime import datetime, timezone
-from pymodbus3.client.sync import ModbusTcpClient
 import uuid
-import copy
+import json
+
+from pymodbus3.client.sync import ModbusTcpClient
 
 from foglamp.common import logger
+from foglamp.plugins.common import utils
 from foglamp.services.south import exceptions
 
 __author__ = "Dan Lopez"
 __copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
+
+""" _DEFAULT_CONFIG with Modbus Entities Map
+
+    The coils and registers each have a read-only table and read-write table.
+
+        Coil	Read-write	1 bit
+        Discrete input	Read-only	1 bit
+        Input register	Read-only	16 bits
+        Holding register	Read-write	16 bits 
+"""
+
 _DEFAULT_CONFIG = {
     'plugin': {
-        'description': 'Python module name of the plugin to load',
-        'type':        'string',
-        'default':     'modbustcp'
+        'description': 'Modbus TCP South Service Plugin',
+        'type': 'string',
+        'default': 'modbustcp'
     },
     'pollInterval': {
-        'description': 'The interval between poll calls to the device poll routine expressed in milliseconds.',
-        'type':        'integer',
-        'default':     '1000'
+        'description': 'The interval between poll calls to the device poll routine, expressed in milliseconds.',
+        'type': 'integer',
+        'default': '1000'
     },
-    #'gpiopin': {
-    'sourceipaddress': {
+    'sourceIpAddress': {
         'description': 'The IP address of the Modbus TCP data source',
-        'type':        'string',
-        'default':     '127.0.0.1'
+        'type': 'string',
+        'default': '127.0.0.1'
+    },
+    'sourcePort': {
+        'description': 'The port of the Modbus TCP data source',
+        'type': 'integer',
+        'default': '502'
+    },
+    'entitiesMap': {

```

Comment:
This does not match the requested name in the Jira ticket and makes it different from the other modbus plugin we have. What not follow the request?

Reply:
we can stick to 
```
"map" : {  "description" : "Modbus register map", 
                        "type" : "JSON", "default" : { 
                                "coils" : { }, 
                                "inputs" : { }, 
                                "registers" : { },
                                "inputRegisters" : { }
                        }
```
if its a MUST need. entitiesMap was kept as map is a inbuilt function name in python.

Also for inputs-> discreteInputs, registers > holdingResgisters are moe standard. Hence I kept that way. I would have point that out once we have final PR. 

anyways, do you think we should stick to map and inputs, registers as key?!